packages/p2p-v2/src/pages/my-profile/screens/MyProfile/MyProfile.tsx

```diff
@@ -14,13 +16,30 @@ const TABS = ['Stats', 'Payment methods', 'Ad details', 'My counterparties'];
 const MyProfile = () => {
     const { isMobile } = useDevice();
     const { queryString, setQueryString } = useQueryString();
+    const { data } = usePoiPoaStatus();
+    const { data: advertiserStats, failureReason, isLoading } = useAdvertiserStats();
+    const { isPoaVerified, isPoiVerified } = data || {};
+    const [isNicknameModalOpen, setIsNicknameModalOpen] = useState<boolean | undefined>(false);
 
     const currentTab = queryString.get('tab');
 
+    useEffect(() => {
+        if (isPoaVerified && isPoiVerified && !!failureReason) setIsNicknameModalOpen(true);
+    }, [failureReason, isPoaVerified, isPoiVerified]);
+
+    if (isLoading && !advertiserStats) {
+        return <Loader />;
+    }
```

Comment:
Instead of individually loading all the components in the my profile component, its better we just have it here for now, since in current behaviour we dont have any skeleton loaders, and we usually load the entire page.

Also needed to check for both isLoading and advertiserStats as the data can be there, but the isLoading can still be true.